### PR TITLE
Random cleanups / improvements to a state

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by gavofyork** [comment]
until we have CI set up to label according to files changed, please label consensus-sensitive PRs as such.

File: ethcore/src/state/mod.rs [hunk 1]
```diff
@@ -853,27 +849,29 @@ impl<B: Backend> State<B> {
 
 	// load required account data from the databases.
 	fn update_account_cache(require: RequireCache, account: &mut Account, state_db: &B, db: &HashDB) {
-		match (account.is_cached(), require) {
```
Comment (rphmeier) [inline ethcore/src/state/mod.rs]: is changing this actually a cleanup or improvement? I think it was fine before

**Comment by rphmeier** [comment]
other changes LGTM

**Comment by gavofyork** [comment]
@debris can you address the grumble?

File: ethcore/src/state/mod.rs (hunk 1 above)
Comment (tomusdrw) [reply ethcore/src/state/mod.rs]: IMHO it's cleaner now (one less identation, clear indication that already cached accounts and `RequireCache::None` is not handled at all).

Comment (debris) [reply ethcore/src/state/mod.rs]: @tomusdrw I was about to write exactly the same thing ;)

**Comment by gavofyork** [comment]
@arkpar review?

**Pull request merged**